src/main/java/sign_up/drivers/SignUpDataAccess.java

```diff
@@ -0,0 +1,207 @@
+    public void saveUser(User toSave) {
+        ArrayList<String[]> userInfo;
+        try {
+            userInfo = readLineByLine(userFilePath);
+        } catch(IOException e) {
+            System.out.println("Wrong Path");
+            userInfo = new ArrayList<>();
+        } catch (CsvValidationException e) {
+            System.out.println("Line Invalid in CSV reader");
+            userInfo = new ArrayList<>();
+        }
+
+        String[] toAdd = {convertListToString((ArrayList<Integer>) toSave.getFavourites()),
+                convertListToString((ArrayList<Integer>) toSave.getPosts()), convertBoolToString(toSave.isAdmin()),
+                String.valueOf(toSave.getId()), toSave.getPassword(), toSave.getEmail()};
+
+        userInfo.add(toAdd);
+
+        try {
+            write(userInfo, userFilePath);
+        } catch (Exception e) {
+            System.out.println("Wrong Path");
+        }
+    }
+
+    @Override
+    public String getAdminPassword() {
+        ArrayList<String[]> userInfo;
+        try {
+            userInfo = readLineByLine(adminFilePath);
+        } catch(IOException e) {
+            System.out.println("Wrong Path");
+            userInfo = new ArrayList<>();
+        } catch (CsvValidationException e) {
+            System.out.println("Line Invalid in CSV reader");
+            userInfo = new ArrayList<>();
+        }
+        return userInfo.get(1)[0];
+    }
+
+    private ArrayList<String[]> readLineByLine(String filePath) throws IOException, CsvValidationException {
+        ArrayList<String[]> list = new ArrayList<>();
+        Reader reader = Files.newBufferedReader(Path.of(filePath));
+        CSVReader csvReader = new CSVReader(reader);
+        String[] line;
+        while ((line = csvReader.readNext()) != null) {
+            list.add(line);
+        }
+        return list;
+    }
+
+    private String convertListToString(ArrayList<Integer> list) {
+        String toReturn = "";
+
+        for (Integer integer: list) {
+            if (list.indexOf(integer) == 0) {
+                toReturn = integer.toString().concat(toReturn);
+            } else {
+                toReturn = integer.toString().concat(".").concat(toReturn);
+            }
+        }
+        return toReturn;
+    }
+
+    private String convertDateToString(LocalDate toConvert) {
+        String toReturn = "";
+        toReturn = toReturn.concat(String.valueOf(toConvert.getYear()));
+        toReturn = toReturn.concat(".");
+        toReturn = toReturn.concat(String.valueOf(toConvert.getMonth()));
+        toReturn = toReturn.concat(".");
+        toReturn = toReturn.concat(String.valueOf(toConvert.getDayOfMonth()));
+
+        return toReturn;
+
+
+    }
+
+    private String convertBoolToString(boolean toConvert) {
+        if (toConvert) {
+            return "1";
+        } else {
+            return "0";
+        }
+    }
+
+    private void write(ArrayList<String[]> toWrite, String filePath) throws Exception {
```

Comment:
Should be more specific as to which exception it throws

Reply:
I think this method has been removed now. I put the uses for it into the specific classes